### DISCUSSION TITLE
Trajectory optimization

### DISCUSSION THREAD

**Discussion by jlmaccal** [cornellius-gp/gpytorch#1626] [Q&A] [answered]
I'm trying to solve a trajectory optimization problem using gpytorch and botorch.

Imagine I have a trajectory of d-dimensional points `x_1, x_2, ..., x_n`.

I have a cost function that is evaluated on adjacent points `c(x_i, x_{i+1}) = z`, where `z` is a scalar. This cost function is stationary in time, so the same function is used to evaluate `c(x_1, x_2)` as for any other `c(x_i, x_{i+1})`. I want to model this function using a GP, which seems straightforward.

What I am having difficulty with is the next part. What I want to do now is to compute the total cost for an entire trajectory as: `c(x_1, x_2) + c(x_2, x_3), + .. C(x_{n-1}, x_n)`. This is also a GP as it's a sum of correlated gaussian random variables. The mean will be the sum of the means of the individual steps and the variance will be the sum over the entire covariance matrix.

However, it isn't clear to me how to best implement this model. Ideally, I would like the resulting model to be treated as a GP, so that I can use analytic acquisition functions. I would also like to make full use of batching for efficiency. Any pointers would be appreciated.

**Comment by Balandat** [comment, marked as answer]
For anyone looking at this, I responded on the botorch clone of this discussion: https://github.com/pytorch/botorch/discussions/805#discussion-3380430
